llvm/test/Transforms/Util/loop-iter-count-assumptions.ll

```diff
@@ -0,0 +1,2326 @@
+; NOTE: Assertions have been autogenerated by utils/update_test_checks.py
+;
+; This file is licensed under the Apache License v2.0 with LLVM Exceptions.
+; See https://llvm.org/LICENSE.txt for license information.
+; SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+;
+; (c) Copyright 2024 Advanced Micro Devices, Inc. or its affiliates
+; unit test for the loop iteration metadata conversion to assumptions
+;
+; NOTE: Example file for converting loop iter count to assumptions in the Loop
+;
+; RUN: opt -S -passes='loop-iter-count-assumptions' -verify-each   < %s | FileCheck %s
+; RUN: opt -S -passes='loop-mssa(licm,loop-iter-count-assumptions)' -verify-each   < %s | FileCheck %s --check-prefix=LICM
```

Comment:
That's a really huge file now. I think the first RUN line isn't all that useful anymore, because your pass is now inserted after LICM in the pass pipeline. Testing it independently outside of its expected conditions doesn't show too much.

Maybe it would make sense to only keep:
 - `opt -S -passes='loop-mssa(licm,loop-iter-count-assumptions)' -verify-each < %s | FileCheck %s --check-prefix=ASSUME`
 - `opt -S -passes='default<O2>' --enable-loop-iter-count-assumptions=1 -start-before='licm' -stop-after='loop-rotate' -verify-each -o - < %s | FileCheck %s --check-prefix=ROTATE`

Nit: I would even only test `loop-mssa(licm,loop-iter-count-assumptions)` in that file to keep it concise and easy to read. Then we can have a separate "integration test" which runs from `licm` to `loop-rotate` with one or two cases. But that's a personal preference at this point :)

Reply:
I wanted to have the end to end test, so that non-obvious errors get shown immediately, i.e. endless loops. 

But sure, now that i am through the testing and know what the expected results are, I can remove it. The integration test is only interesting for edge cases, which would duplicates large parts of the unittest, which is the original reason I had them in the same file.